[rfc] wpa_supplicant: use capabilities in service to run as non-root

dkwo opened the pull request:
The capabilities are the ones recommended in upstream README. They recommend to set them on the executable, but it seems safer to set them in the service.

- I tested the changes in this PR: yes, but only using the runit service. I.e., did not test other potential consumers of wpa_supplicant, like network manager or wpa_cli. 
- I built this PR locally for my native architecture, (x86_64-glibc)

Added a patch to the external file function, which is now upstream.

dkwo commented:
@Gottox @leahneukirchen What do you think?

leahneukirchen commented:
Perhaps we could `chown` the config file in the `run` script, so it would automigrate?

dkwo commented:
That's an option. Note that, if `control_interface_group` is not set to `_wpas`, the daemon will not work iirc, as it needs to create stuff with the right user and group permissions. If one has a custom `wpa_supplicant.conf`, they should be warned by an INSTALL.msg to change their conf file.

I added a post-install trigger and message, but I can change to do it in the service if that's preferred.
Just rebased and split the commits.

vulpes-vulpeos commented:
> ...this has been running flawlessly for me since November.
It completely broke wpa_cli utility for me. I get this warning:
```
Warning: Failed to attach to wpa_supplicant.
Could not connect to wpa_supplicant: wlp1s0 - re-trying
```
I changed control_interface_group to _wpas  in /etc/wpa_supplicant/wpa_supplicant.conf and added user into _wpas group.
Also running wpa_cli with sudo doesn't change anything.
What do I need to do to fix it?

vulpes-vulpeos commented:
```
sudo setpriv --reuid _wpas --regid _wpas --clear-groups wpa_cli 
sudo -u _wpas wpa_cli
```
Thank you, both of this worked, but I hope there will be permanent solution.

ahesford commented:
@dkwo Do you have a proper fix here? If not, let's revert this before more people are bitten.
